src/persist-client/src/internal/state_versions.rs

```diff
@@ -1086,21 +1085,26 @@ impl<T: Timestamp + Lattice + Codec64> ReferencedBlobValidator<T> {
             }
         });
 
-        if let Some(first_incr) = self.inc_batches.first() {
-            let first_full = self
-                .full_batches
-                .first()
-                .expect("full has at least 1 batch");
-            assert_eq!(first_incr.desc.lower(), first_full.desc.lower());
-            PartialOrder::less_equal(first_full.desc.since(), first_incr.desc.since());
```

Comment:
@pH14 - I know this is a deep cut, but do you recall what these `less_equal` calls were for? (I assume we meant to assert on the result, but I don't know why we'd expect the full-batch since to be less than the incremental-batch since.)

Reply:
Shouldn't the lowest and highest sinces we see exactly match between "full" and "incremental"? We could also collect the lowest (or meet) and highest (or join) in a loop and add an assertion, so add them to the result type of `overall_desc()`?